docs/content/docs/concepts/glossary.md

```diff
@@ -25,182 +25,605 @@ under the License.
+because a bounded data set can always be sorted. Processing of bounded streams is also known as 
+[batch processing](#batch-processing).
+
+#### Checkpoint
+
+A [snapshot](#snapshot) taken automatically by Flink for the purpose of being able to recover from 
+faults. A checkpoint marks a specific point in each of the input streams along with the corresponding 
+state for each of the operators. Checkpoints can be incremental and unaligned, and are optimized for 
+being restored quickly.
+
+#### Checkpoint Barrier
+
+A special marker that flows along the graph and triggers the checkpointing process on each of the 
+parallel instances of the operators. Checkpoint barriers are injected into the source operators and 
+flow together with the data. If an operator has multiple outputs, it gets "split" into both of them.
+
+#### Checkpoint Coordinator
+
+This coordinates the distributed snapshots of operators and state. It is part of the JobManager and 
+instructs the TaskManager when to begin a checkpoint by sending the messages to the relevant tasks 
+and collecting the checkpoint acknowledgements.
+
 #### Checkpoint Storage
 
-The location where the [State Backend](#state-backend) will store its snapshot during a checkpoint (Java Heap of [JobManager](#flink-jobmanager) or Filesystem).
+The location where the [state backend](#state-backend) will store its snapshot during a checkpoint. 
+This could be on the Java heap of the [JobManager](#flink-jobmanager) or on a file system.
+
+#### (Flink) Client
+
+This is not part of the runtime and program execution but is used to prepare and send a dataflow graph 
+to the JobManager. The Flink client runs either as part of the program that triggers the execution or 
+in the command line process via `./bin/flink run`.
+
+#### (Flink) Cluster
 
-#### Flink Application Cluster
+A distributed system consisting of (typically) one [JobManager](#jobmanager) and one or more
+[TaskManager](#taskmanager) processes.
 
-A Flink Application Cluster is a dedicated [Flink Cluster](#flink-cluster) that
-only executes [Flink Jobs](#flink-job) from one [Flink
-Application](#flink-application). The lifetime of the [Flink
-Cluster](#flink-cluster) is bound to the lifetime of the Flink Application.
+#### Connected Streams
 
-#### Flink Job Cluster
+A pattern in Flink where a single operator has two input streams. Connected streams can also be used 
+to implement streaming joins.
 
-A Flink Job Cluster is a dedicated [Flink Cluster](#flink-cluster) that only
-executes a single [Flink Job](#flink-job). The lifetime of the
-[Flink Cluster](#flink-cluster) is bound to the lifetime of the Flink Job. 
-This deployment mode has been deprecated since Flink 1.15.  
+#### Connectors
 
-#### Flink Cluster
+Connectors allow [Flink applications](#(flink)-applications) to read from and write to various external 
+systems. They support multiple formats in order to encode and decode data to match Flink’s data structures.
```

Comment:
Please link to sink and source. We often use the terms intercheangeably, or better we often use the term "connector" as the umbrella term to talk about a sink/source implementation for a specific external system. For example the kafka connector is the sink and source implementation for kafka.